app/upgrade-poolChange.go

```diff
@@ -0,0 +1,96 @@
+package app
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Sifchain/sifnode/x/clp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/upgrade"
+	"io/ioutil"
+)
+
+func GetPoolChangeFunc(app *SifchainApp) func(ctx sdk.Context, plan upgrade.Plan) {
+	return func(ctx sdk.Context, plan upgrade.Plan) {
+		ctx.Logger().Info("Starting to execute upgrade plan for pool re-balance")
+
+		ExportAppState("changePoolFormula", app, ctx)
+
+		allPools := app.clpKeeper.GetPools(ctx)
+		lps := clp.LiquidityProviders{}
+		poolList := clp.Pools{}
+		hasError := false
+		for _, pool := range allPools {
+			lpList := app.clpKeeper.GetLiquidityProvidersForAsset(ctx, pool.ExternalAsset)
+			temp := sdk.ZeroUint()
+			tempExternal := sdk.ZeroUint()
+			tempNative := sdk.ZeroUint()
+			for _, lp := range lpList {
+				withdrawNativeAssetAmount, withdrawExternalAssetAmount, _, _ := clp.CalculateWithdrawal(pool.PoolUnits, pool.NativeAssetBalance.String(),
+					pool.ExternalAssetBalance.String(), lp.LiquidityProviderUnits.String(), sdk.NewUint(clp.MaxWbasis).String(), sdk.NewInt(0))
```

Comment:
nit: a bit difficult to read
```suggestion
				withdrawNativeAssetAmount, withdrawExternalAssetAmount, _, _ := clp.CalculateWithdrawal(
				  pool.PoolUnits, 
				  pool.NativeAssetBalance.String(),
				  pool.ExternalAssetBalance.String(), 
				  lp.LiquidityProviderUnits.String(), 
				  sdk.NewUint(clp.MaxWbasis).String(), 
				  sdk.NewInt(0),
				)
```